var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -145,373 +145,71 @@ class IntelParallelStudio(IntelPackage):
-            spack_env.set('I_MPI_CXX', spack_cxx)
-            spack_env.set('I_MPI_F77', spack_fc)
-            spack_env.set('I_MPI_F90', spack_f77)
-            spack_env.set('I_MPI_FC',  spack_fc)
-
-        # set up MKLROOT for everyone using MKL package
-        if '+mkl' in self.spec:
-            mkl_root = self.prefix.compilers_and_libraries.linux.mkl.lib.intel64  # noqa
-
-            spack_env.set('MKLROOT', self.prefix)
-            spack_env.append_path('SPACK_COMPILER_EXTRA_RPATHS', mkl_root)
-
-    def setup_dependent_package(self, module, dep_spec):
-        if '+mpi' in self.spec:
-            # Intel comes with 2 different flavors of MPI wrappers:
-            #
-            # * mpiicc, mpiicpc, and mpifort are hardcoded to wrap around
-            #   the Intel compilers.
-            # * mpicc, mpicxx, mpif90, and mpif77 allow you to set which
-            #   compilers to wrap using I_MPI_CC and friends. By default,
-            #   wraps around the GCC compilers.
-            #
-            # In theory, these should be equivalent as long as I_MPI_CC
-            # and friends are set to point to the Intel compilers, but in
-            # practice, mpicc fails to compile some applications while
-            # mpiicc works.
-            bindir = self.prefix.compilers_and_libraries.linux.mpi.intel64.bin
-
-            if self.compiler.name == 'intel':
-                self.spec.mpicc  = bindir.mpiicc
-                self.spec.mpicxx = bindir.mpiicpc
-                self.spec.mpifc  = bindir.mpiifort
-                self.spec.mpif77 = bindir.mpiifort
-            else:
-                self.spec.mpicc  = bindir.mpicc
-                self.spec.mpicxx = bindir.mpicxx
-                self.spec.mpifc  = bindir.mpif90
-                self.spec.mpif77 = bindir.mpif77
-
-    def setup_environment(self, spack_env, run_env):
-        """Adds environment variables to the generated module file.
-
-        These environment variables come from running:
-
-        .. code-block:: console
-
-           $ source parallel_studio_xe_2017/bin/psxevars.sh intel64
-        """
-        # NOTE: Spack runs setup_environment twice, once pre-build to set up
-        # the build environment, and once post-installation to determine
-        # the environment variables needed at run-time to add to the module
-        # file. The script we need to source is only present post-installation,
-        # so check for its existence before sourcing.
-        # TODO: At some point we should split setup_environment into
-        # setup_build_environment and setup_run_environment to get around
-        # this problem.
-        psxevars = glob.glob(join_path(
-            self.prefix, 'parallel_studio*', 'bin', 'psxevars.sh'))
-
-        if psxevars:
-            run_env.extend(EnvironmentModifications.from_sourcing_file(
-                psxevars[0], 'intel64'))
+# TODO: I don't think this still applies as of 2017.3 and later:
```

Comment:
Do we need this comment block? It looks like the version check below already addresses the fix in 2017.3 and greater.

Reply:
Good catch. Actually, the conditional will have to read:
``` python
if self.version_yearlike.satisfies(ver('2016.0:2017.2')):
```
with `_version_yearlike` from `lib/spack/spack/build_systems/intel.py` shedding its `_` prefix.